### PR TITLE
Add resource api json test base and paragraph json api test case

### PR DESCRIPTION
**Pull request by tannguyen04**

### CODE REVIEW

**Comment by tannguyen04** [comment]
Hi @simesy 
In this PR I have done:
1. Create resource test base class.
2. Create resource paragraph html (html component) test class.
Could you please review and give feedback. thank you.

File: web/modules/custom/lecapi/tests/src/Functional/LecapiResourceParagraphHtmlTest.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+          'format' => 'basic',
+        ],
+        'subtitle' => $subtitle_paragraph,
+      ]);
+    /** @var \Drupal\node\Entity\Node $page_node */
+    $page_node = $this->entityTypeManager
+      ->getStorage('node')
+      ->create([
+        'type' => 'page',
+        'content' => [
```
Comment (simesy) [inline web/modules/custom/lecapi/tests/src/Functional/LecapiResourceParagraphHtmlTest.php:50]: Should be a constant `Ia::FIELD_CONTENT`

File: web/modules/custom/lecapi/tests/src/Functional/LecapiResourceParagraphHtmlTest.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+          $html_paragraph,
+        ],
+        'title' => 'Test HTML component',
+        'uid' => $this->account->id(),
+        Ia::FIELD_SITE => $this->getSiteTerm(),
+      ]);
+    $page_node->save();
+    // Set html paragraph as entity be test.
+    $referenced_entities = $page_node->get('content')->referencedEntities();
```
Comment (simesy) [inline web/modules/custom/lecapi/tests/src/Functional/LecapiResourceParagraphHtmlTest.php:59]: `Ia::FIELD_CONTENT`

File: web/modules/custom/lecapi/tests/src/LecapiResourceTestBase.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    $request_options = [];
+    $request_options[RequestOptions::HEADERS]['Accept'] = 'application/vnd.api+json';
+    $this->drupalLogin($this->account);
+    $response = $this->request('GET', $url, $request_options);
+    $expected_document = $this->getExpectedResponse();
+    // Assert response code.
+    $this->assertEqual($response->getStatusCode(), 200);
+    // Assert response body.
+    $actual_document = Json::decode($response->getBody()->__toString());
+    $this->assertArraySimilar($expected_document, $actual_document);
```
Comment (simesy) [inline web/modules/custom/lecapi/tests/src/LecapiResourceTestBase.php:58]: I think this is TOO smart. Would it be a bit clearer just to check the important values and then we don't really care about the other stuff that JSONAPI returns:
```
$this->assertSame($actual_document['return']['data']['attributes']['_markup']['value'], '<p>This is text markup</p>');
```

**Review by simesy — COMMENTED** [review]
This is pretty fancy... I think too fancy. I'm going to think about it.

File: web/modules/custom/lecapi/tests/src/LecapiResourceTestBase.php (hunk 3 above)
Comment (simesy) [reply web/modules/custom/lecapi/tests/src/LecapiResourceTestBase.php:58]: `$this->assertArraySimilar(...` feels like "We are going to do this very complex check, and we don't mind if it fails a little bit" :D

**Review by simesy — APPROVED** [review]

**Pull request merged**